packages/account/src/providers/transaction-request/transaction-request.ts

```diff
@@ -249,6 +265,32 @@ export abstract class BaseTransactionRequest implements BaseTransactionRequestLi
     this.witnesses[index] = witness;
   }
 
+  /**
+   * Helper function to add an external signer to the transaction.
+   *
+   * @param signer - The signer to add to the transaction.
+   * @returns The transaction with the signature witness added.
+   */
+  async addSigner(signer: Account) {
```

Comment:
The name of the method `addSigner` doesn't fully carry the meaning of the implementation. It opens up additional questions like "What happens with the signer when I add it?", "Will it be used later for something?", "Does the signer sign the transaction immediately?" - at least for me.

I believe a better approach would be to expand `addWitness` and remove this method:
```ts
  /**
   * Adds a witness to the request.
   * If the value is a `BaseWalletUnlocked`, the wallet signs the transaction and the signature is pushed into the witnesses.
   * If the value is `BytesLike`, it gets directly pushed into the witnesses.
   *
   * @param signerOrSignature - The signature to add to the witness.
   * @returns The index of the created witness.
   */
  addWitness(signerOrSignature: BaseWalletUnlocked | BytesLike) {
    if(signerOrSignature instanceof BaseWalletUnlocked) {
        const hashedTransaction = this.getTransactionId(signer.provider.getChainId());
        const signature = await signer.signTransaction(hashedTransaction);
        this.witnesses.push(signature);
    } else {
        this.witnesses.push(signerOrSignature);
    }

    return this.witnesses.length - 1;
  }
```

Reply:
Both singular and plural alternatives work fine, but I'd be cautious with type overloading. Although `string | string[]` is obvious and perfectly fine, I'm usually [against](https://github.com/FuelLabs/fuels-ts/issues/748#issuecomment-1400789719) overloading completely different types like `Account | BytesLike`, as abstractions start to creep in. 

@nedsalk The same questions could arise regardless of the terminology.

Reply:
@danielbate and I had a chat around it and opted for a separate method instead of overloading `addWitness`:
```ts
addAccountWitness(account: Account) {...}
```
or something like that.